lfs.c

```diff
@@ -4792,22 +4798,15 @@ static lfs_stag_t lfs_fs_parent(lfs_t *lfs, const lfs_block_t pair[2],
     // use fetchmatch with callback to find pairs
     parent->tail[0] = 0;
     parent->tail[1] = 1;
-    lfs_block_t tortoise[2] = {LFS_BLOCK_NULL, LFS_BLOCK_NULL};
-    lfs_size_t tortoise_i = 1;
-    lfs_size_t tortoise_period = 1;
+    struct lfs_tortoise_t tortoise = {
+        .pair = {LFS_BLOCK_NULL, LFS_BLOCK_NULL},
+        .i = 1,
+        .period = 1,
+    };
     while (!lfs_pair_isnull(parent->tail)) {
-        // detect cycles with Brent's algorithm
-        if (lfs_pair_issync(parent->tail, tortoise)) {
-            LFS_WARN("Cycle detected in tail list");
+        if (lfs_detect_cycles(parent, &tortoise) == LFS_ERR_CORRUPT) {
```

Comment:
This should probably be:
``` c
int err = lfs_detect_cycles(parent, &tortoise);
if (err) {
    return err;
}
```

For the above reason, and for consistency with the codebase. Putting non-trivial functions inside if/while statements make them easy to miss when reading the code.